recipes/liquid-dsp/all/conanfile.py

```diff
@@ -0,0 +1,178 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from contextlib import contextmanager
+import os
+import glob
+import copy
+
+
+class LiquidDspConan(ConanFile):
+    name = "liquid-dsp"
+    description = "Digital signal processing library for software-defined radios "
+    topics = ("conan", "dsp", "sdr", "liquid-dsp")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/jgaeddert/liquid-dsp"
+    license = ("MIT",)
+    settings = "os", "arch", "build_type", "compiler"
+    exports_sources = ["generate_link_library.bat", "patches/**"]
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "simdoverride": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "simdoverride": False,
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _libname(self):
+        if self.settings.os == "Windows":
+            return "libliquid"
+        return "liquid"
+
+    @property
+    def _target_name(self):
+        if self.settings.os == "Macos":
+            if not self.options.shared:
+                return "libliquid.ar"
+            return "libliquid.dylib"
+        if not self.options.shared:
+            return "libliquid.a"
+        return "libliquid.so"
+
+    @property
+    def _lib_pattern(self):
+        if self.settings.os == "Macos" and not self.options.shared:
+            return "libliquid.a"
+        if self.settings.os != "Windows":
+            return self._target_name
+        return "libliquid.lib"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os != "Windows":
+            if not self.options.fPIC:
+                raise ConanInvalidConfiguration("This library hardcodes fPIC")
```

Comment:
```suggestion
        if self.settings.os != "Windows":
            if not self.options.fPIC:
                raise ConanInvalidConfiguration("This library hardcodes fPIC")
```

Two options here: Or, you provides a patch which allows fPIC, or you just remove fPIC from the recipe as it's not optional.

Reply:
I thought fPIC was mandatory. Will do!